### PR TITLE
DYN-7112: Fix DS parser to error early on finding import statements in code block nodes

### PR DESCRIPTION
**Pull request by aparajit-pratap**
### Purpose

Fix DS parser to error early on finding import statements in code block nodes

### Declarations

Check these if you believe they are true

- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated
- [ ] This PR contains no files larger than 50 MB

### Release Notes

Fix parser to error early on finding import statements in code block nodes

### CODE REVIEW

**Review by mjkkirschner — APPROVED** [review]

**Comment by aparajit-pratap** [comment]
> any way to unit test this so no one breaks it later?

Sure, will add one.

I also need to fix failing tests as they run under the assumption that `IsParsingCodeBlockNode` is true while at the same time containing import statements. As this is not a real Dynamo use-case, the tests/test framework needs some adjustment as to not make these assumptions.

File: src/Engine/ProtoCore/Parser/atg/Associative.atg [hunk 1]
```diff
@@ -2028,9 +2034,6 @@ Associative_IdentifierList<out ProtoCore.AST.AssociativeAST.AssociativeNode node
                                             }
                                             else
                                             {
-                                                string rhsName = null;
```
Comment (mjkkirschner) [inline src/Engine/ProtoCore/Parser/atg/Associative.atg:2031]: unused?

File: src/Engine/ProtoCore/Utils/ParserUtils.cs [hunk 2]
```diff
@@ -140,8 +140,8 @@ private static Core CodeBlockParserCore()
             var core = new Core(new Options())
             {
                 ParsingMode = ParseMode.AllowNonAssignment,
-                IsParsingCodeBlockNode = true,
-                IsParsingPreloadedAssembly = false
+                IsParsingPreloadedAssembly = false,
+                IsParsingInTestMode = true,
```
Comment (mjkkirschner) [inline src/Engine/ProtoCore/Utils/ParserUtils.cs:144]: why should this be true here?

File: test/Engine/ProtoTest/LiveRunnerTests/MicroFeatureTests.cs [hunk 3]
```diff
@@ -644,18 +642,18 @@ public void TestDeltaExpression_01()
         [Test]
         public void TestDeltaExpression_02()
         {
-            liveRunner = new ProtoScript.Runners.LiveRunner();
+            //liveRunner = new ProtoScript.Runners.LiveRunner();
```
Comment (mjkkirschner) [inline test/Engine/ProtoTest/LiveRunnerTests/MicroFeatureTests.cs:645]: can this be removed?

File: src/Engine/ProtoCore/Parser/Parser.cs [hunk 4]
```diff
@@ -791,7 +791,7 @@ bool WeakSeparator(int n, int syFol, int repFol) {
 	void DesignScriptParser() {
 		Node node = null; 
 		Hydrogen(out node);
-		if (!core.IsParsingPreloadedAssembly && !core.IsParsingCodeBlockNode && !builtinMethodsLoaded)
+		if (!core.IsParsingPreloadedAssembly && !core.IsParsingCodeBlockNode && !builtinMethodsLoaded && !core.IsParsingInTestMode)
```
[the remark this answers is not in the document]
Comment (aparajit-pratap) [reply src/Engine/ProtoCore/Parser/Parser.cs:794]: It's been edited in start.atg and regenerated.

File: src/Engine/ProtoCore/Parser/atg/Associative.atg (hunk 1 above)
Comment (aparajit-pratap) [reply src/Engine/ProtoCore/Parser/atg/Associative.atg:2031]: Yes

File: src/Engine/ProtoCore/Utils/ParserUtils.cs (hunk 2 above)
Comment (aparajit-pratap) [reply src/Engine/ProtoCore/Utils/ParserUtils.cs:144]: It's actually a test helper. I'll see if I can refactor it or rename it so that's clear.

**Review by mjkkirschner — APPROVED** [review]

File: test/Engine/ProtoTest/FFITests/CSFFIDispose.cs [hunk 5]
```diff
@@ -2907,14 +2907,6 @@ public void DisposeEnumWrapper()
             thisTest.Verify("x", Days.Monday);
         }
 
-        //Migrate this code into the test framework
-        private Subtree CreateSubTreeFromCode(Guid guid, string code)
-        {
-            var cbn = ProtoCore.Utils.ParserUtils.Parse(code);
-            var subtree = null == cbn ? new Subtree(null, guid) : new Subtree(cbn.Body, guid);
-            return subtree;
-        }
```
Comment (aparajit-pratap) [inline test/Engine/ProtoTest/FFITests/CSFFIDispose.cs:2916]: Unused

File: test/Engine/ProtoTest/LiveRunnerTests/ChangeSetComputerTests.cs [hunk 6]
```diff
@@ -24,13 +24,6 @@ public override void Setup()
             runtimeCore = new RuntimeCore(new Heap());
         }
 
-        private Subtree CreateSubTreeFromCode(Guid guid, string code)
-        {
-            var cbn = ProtoCore.Utils.ParserUtils.Parse(code);
-            var subtree = null == cbn ? new Subtree(null, guid) : new Subtree(cbn.Body, guid);
-            return subtree;
-        }
```
Comment (aparajit-pratap) [inline test/Engine/ProtoTest/LiveRunnerTests/ChangeSetComputerTests.cs:32]: Unused

**Pull request merged**